### PR TITLE
Attempted fix for flashing while automatically switching chain

### PR DESCRIPTION
**Pull request by mudrila**
Pass success and error callback, use plain switchChain in DAOController

### CODE REVIEW

**Comment by mudrila** [comment]
@decentdao/engineering Another possible improvement to the UI / UX would be:

1. Adding Network selector to the `We couldn't find a Safe at this address on ${chain.name}` error
2. When switching chain through that network select - change network prefix in the URL to attempt loading Safe on different network

Other ideas
- Attempt to fetch Safe from all the other supported networks to identify whether there's a matching Safe on other network. Then show button to switch to proper chain OR toast with action to switch to proper chain OR automatically switch to the proper chain, if there's only 1 matching Safe 

@decentdao/design Also would like your take on the above message

**Comment by nicolaus-sherrill** [comment]
> @decentdao/engineering Another possible improvement to the UI / UX would be:
> 
> 1. Adding Network selector to the `We couldn't find a Safe at this address on ${chain.name}` error
> 
> Screenshot 2024-10-14 at 16 33 41 2. When switching chain through that network select - change network prefix in the URL to attempt loading Safe on different network
> Other ideas
> 
> * Attempt to fetch Safe from all the other supported networks to identify whether there's a matching Safe on other network. Then show button to switch to proper chain OR toast with action to switch to proper chain OR automatically switch to the proper chain, if there's only 1 matching Safe
> 
> @decentdao/design Also would like your take on the above message

hmmmm

I like your last bullet a lot. It's the most helpful to the user to get them where they need to be, without requiring too much action or disrupting their flow too much.

Is it possible to load the Safe when connected to the wrong network? Rather than throwing an error here, we could show the Safe and overlay the UI with a "Switch Network" CTA.

**Review by adamgall — APPROVED** [review]

File: src/i18n/locales/en/common.json [hunk 1]
```diff
@@ -115,5 +115,6 @@
   "goTo": "Go to",
   "none": "None",
   "ipfsLoadingErrorMessage": "Error fetching data from IPFS, please try again later.",
-  "ipfsSavingErrorMessage": "Error saving data to IPFS, please try again later."
+  "ipfsSavingErrorMessage": "Error saving data to IPFS, please try again later.",
+  "automaticChainSwitchingErrorMessage": "We coulnd't automatically switch network to the Safe's netowrk. Please, try to switch network in your wallet application."
```
[the remark this answers is not in the document]
Comment (mudrila) [reply src/i18n/locales/en/common.json]: Fixed, thank you!

**Review by Da-Colon — COMMENTED** [review]
So flashing is gone, but When I switch networks say mainnet -> sepolia. I get a second pop up sepolia -> mainnet

**Comment by mudrila** [comment]
> So flashing is gone, but When I switch networks say mainnet -> sepolia. I get a second pop up sepolia -> mainnet

You mean it is not preventing automatic switch when you "switch back" (or to another chain that is not in the prefix)?

Yeah, that's kind of a problem I'm trying to address with suggestions I've posted previously. With the UI/UX that @nicolaus-sherrill suggested - this problem will be solved, I believe

**Comment by Da-Colon** [comment]
> > So flashing is gone, but When I switch networks say mainnet -> sepolia. I get a second pop up sepolia -> mainnet
> 
> You mean it is not preventing automatic switch when you "switch back" (or to another chain that is not in the prefix)?
> 
> Yeah, that's kind of a problem I'm trying to address with suggestions I've posted previously. With the UI/UX that @nicolaus-sherrill suggested - this problem will be solved, I believe

Ah actually it happen on dev and it seems unrelated. If you go to a DAO via direct link. then back to home page and switch to a different network. the popup opens up twice, once to the network you want to change to and the second one back. weird

**Review by Da-Colon — APPROVED** [review]

**Pull request merged**